lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/client/REEFImplementation.java

```diff
@@ -89,8 +90,21 @@
 
   @Override
   public void close() {
+
+    LOG.log(Level.FINE, "Close REEF: shutdown jobs");
     this.runningJobs.closeAllJobs();
+
+    LOG.log(Level.FINE, "Close REEF: shutdown client");
     this.clientWireUp.close();
+
+    LOG.log(Level.FINE, "Close REEF: shutdown job submitter");
+    try {
+      this.jobSubmissionHandler.close();
```

Comment:
@motus gentle ping :) Have you succeded to reproduce the sporadic event?

@tcNickolas @jwang98052 In my opinion, we don't have to skip `jobSubmissionHandler.close()`, though we haven't seen any problemetic case without calling it. Is there anything I'm missing?